conformance/plugins/osm-arc/osm_arc_conformance.sh

```diff
@@ -24,9 +24,76 @@ function waitForResources {
     echo "$available"
 }
 
+testsRun="false"
 # saveResults prepares the results for handoff to the Sonobuoy worker.
 # See: https://github.com/vmware-tanzu/sonobuoy/blob/master/docs/plugins.md
 saveResults() {
+
+  if [[ "$testsRun" == "true" ]]; then
+    results=$(cat ./temp_results/results.xml)
+
+    echo $results
+
+    if [[ "$results" =~ .*"Summarizing "+[0-9]*+" Failures:" ]]; then
+        # Parse xml output to get all tests that are failing
+        failed_tests=$(echo $results | awk -F 'Summarizing ' '{print $2}')
+        IFS='/'
+
+        # Split into array with '/' as a delimiter
+        read -a failed_tests_arr <<< "$failed_tests"
+
+        test_names=()
+        for val in "${failed_tests_arr[@]}"; do
+            # Look for "[]", "[Cross-platform]", or "[linux]", and get the test name from rest of substring. 
+            if [[ "$val" == *"[]"* ]]; then
+                temp_1=$(echo $val | awk -F '\\[\\] ' '{print $2}')
+                temp_2=$(echo $temp_1 | awk -F '\\[' '{print $1}')
+                test_name=$(echo ${temp_2% *})
+                test_name_clean=$(echo "${test_name//&gt;/>}")
+                test_names+=("$test_name_clean")
+            fi
+            if [[ "$val" == *"[Cross-platform]"* ]]; then
+                temp_1=$(echo $val | awk -F '\\[Cross-platform\\] ' '{print $2}')
+                temp_2=$(echo $temp_1 | awk -F '\\[' '{print $1}')
+                test_name=$(echo ${temp_2% *})
+                test_name_clean=$(echo "${test_name//&gt;/>}")
+                test_names+=("$test_name_clean")
+            fi
+            if [[ "$val" == *"[linux]"* ]]; then
+                temp_1=$(echo $val | awk -F '\\[linux\\] ' '{print $2}')
+                temp_2=$(echo $temp_1 | awk -F '\\[' '{print $1}')
+                test_name=$(echo ${temp_2% *})
+                test_name_clean=$(echo "${test_name//&gt;/>}")
+                test_names+=("$test_name_clean")
+            fi
```

Comment:
I'm not super familiar with scripting but could this be simplified since most of the logic within the conditions is the same? something like (pseudocode):

if val == [] or [cross-platform] or [linux] then:
   temp_1=$(echo $val | awk -F '\\[$val] ' '{print $2}')

and then share the rest of the logic?

Reply:
Good catch! I think the "temp_1" logic would still need to be separate for each case, because of the way the brackets are escaped (using $var) won't quite work, but the rest of the logic can be shared.